tests/tests/test-fees/test-fee-multiplier.ts

```diff
@@ -160,3 +156,239 @@ describeDevMoonbeam("Max Fee Multiplier - initial value", (context) => {
     expect(initialValue).to.equal(8_000_000_000_000_000_000n);
   });
 });
+
+describeDevMoonbeam("Fee Multiplier - XCM Executions", (context) => {
+  const startingBn = new BN("2000000000000000000");
+  let sendingAddress: string;
+  let random: KeyringPair;
+  let transferredBalance: bigint;
+  let balancesPalletIndex: number;
+
+  before("Suite Setup", async function () {
+    const { originAddress, descendOriginAddress } = descendOriginFromAddress(context);
+    sendingAddress = originAddress;
+    random = generateKeyringPair();
+    transferredBalance = 10_000_000_000_000_000_000n;
+
+    await expectOk(
+      context.createBlock(
+        context.polkadotApi.tx.balances.transfer(descendOriginAddress, transferredBalance * 100n)
+      )
+    );
+
+    const metadata = await context.polkadotApi.rpc.state.getMetadata();
+    balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find((pallet) => {
+      return pallet.name === "Balances";
+    }).index;
+  });
+
+  beforeEach("Reset multiplier", async function () {
+    const MULTIPLIER_STORAGE_KEY = context.polkadotApi.query.transactionPayment.nextFeeMultiplier
+      .key(0)
+      .toString();
+
+    await context.polkadotApi.tx.sudo
+      .sudo(
+        context.polkadotApi.tx.system.setStorage([
+          [MULTIPLIER_STORAGE_KEY, bnToHex(startingBn, { isLe: true, bitLength: 128 })],
+        ])
+      )
+      .signAndSend(alith);
+    await context.createBlock();
+  });
+
+  it("should decay with no activity", async function () {
+    const initialValue = await context.polkadotApi.query.transactionPayment.nextFeeMultiplier();
+    expect(initialValue.lt(startingBn), "Fee Multiplier value not decayed").to.be.true;
```

Comment:
I think this test should create a block to be more clear. There are minimum and maximum multipliers that will be enforced and it can be a bit subtle as to when they kick in (at the beginning or end of a block, for example.) So it's not clear exactly how this test expects that to occur.

Reply:
Dpme